model/adjuster/span_merger_test.go

```diff
@@ -0,0 +1,345 @@
+							{
+								RefType: model.SpanRefType_CHILD_OF,
+								TraceID: model.NewTraceID(2, 3),
+								SpanID:  model.SpanID(4),
+							},
+						},
+					},
+					{
+						SpanID:     model.SpanID(1),
+						Duration:   5 * time.Microsecond,
+						Warnings:   []string{"Third Warning"},
+						Incomplete: true,
+					},
+					{
+						SpanID:   model.SpanID(1),
+						Duration: 8 * time.Microsecond,
+					},
+					{
+						SpanID: model.SpanID(2),
+					},
+				},
+			},
+			expected: &model.Trace{
+				Spans: []*model.Span{
+					{
+						SpanID:   model.SpanID(1),
+						Duration: 8 * time.Microsecond,
+						References: []model.SpanRef{
+							{
+								RefType: model.SpanRefType_CHILD_OF,
+								TraceID: model.NewTraceID(2, 3),
+								SpanID:  model.SpanID(3),
+							},
+							{
+								RefType: model.SpanRefType_CHILD_OF,
+								TraceID: model.NewTraceID(2, 3),
+								SpanID:  model.SpanID(4),
+							},
+						},
+						Warnings: []string{"First Warning", "Second Warning", "Third Warning"},
+					},
+					{
+						SpanID: model.SpanID(2),
+					},
+				},
+			},
+		},
+		{
+			name: "duplicate Jaeger spans: properly merge incomplete spans if no final span is stored",
+			input: &model.Trace{
+				Spans: []*model.Span{
+					{
+						SpanID:     model.SpanID(1),
+						Incomplete: true,
+						Duration:   1 * time.Microsecond,
+						Warnings:   []string{"First Warning", "Second Warning"},
+					},
+					{
+						SpanID: model.SpanID(1),
+						References: []model.SpanRef{
+							{
+								RefType: model.SpanRefType_CHILD_OF,
+								TraceID: model.NewTraceID(2, 3),
+								SpanID:  model.SpanID(3),
+							},
+						},
+						Duration:   2 * time.Microsecond,
+						Incomplete: true,
+					},
+					{
+						SpanID:     model.SpanID(1),
+						Duration:   5 * time.Microsecond,
+						Incomplete: true,
+						References: []model.SpanRef{
+							{
+								RefType: model.SpanRefType_CHILD_OF,
+								TraceID: model.NewTraceID(2, 3),
+								SpanID:  model.SpanID(4),
+							},
+						},
+					},
+					{
+						SpanID:     model.SpanID(1),
+						Duration:   6 * time.Microsecond,
+						Warnings:   []string{"Third Warning"},
+						Incomplete: true,
+					},
+					{
+						SpanID: model.SpanID(2),
+					},
+				},
+			},
+			expected: &model.Trace{
+				Spans: []*model.Span{
+					{
+						SpanID:     model.SpanID(1),
+						Duration:   6 * time.Microsecond,
+						Incomplete: true,
+						References: []model.SpanRef{
+							{
+								RefType: model.SpanRefType_CHILD_OF,
+								TraceID: model.NewTraceID(2, 3),
+								SpanID:  model.SpanID(3),
+							},
+							{
+								RefType: model.SpanRefType_CHILD_OF,
+								TraceID: model.NewTraceID(2, 3),
+								SpanID:  model.SpanID(4),
+							},
+						},
+						Warnings: []string{"First Warning", "Second Warning", "Third Warning"},
+					},
+					{
+						SpanID: model.SpanID(2),
+					},
+				},
+			},
+		},
+		{
+			name: "duplicate Zipkin spans: don't merge",
```

Comment:
I would prefer this to be (a) a separate test, and (b) use the actual Zipkin format (e.g. zipkin.thrift) as the input, with the full chain of conversion & adjusters, especially including the deduper.